atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/TaskRunner.java

```diff
@@ -44,22 +50,43 @@
             }
         }
 
-        List<Future<V>> futures = new ArrayList<>(tasks.size());
+        List<ListenableFuture<V>> futures = new ArrayList<>(tasks.size());
         for (Callable<V> task : tasks) {
-            futures.add(executor.submit(task));
+            DetachedSpan detachedSpan = DetachedSpan.start("task");
```

Comment:
not actionable now: Yep, this is a reasonable first step. Later on we probably want a bit more information about what the tasks are doing (i.e. maybe change the method's signature or also allow users to provide a List<TaskContext> where that has metadata about th task), but this would give us some useful signal.

Reply:
agreed